javalib/src/main/scala/java/util/ArrayDeque.scala

```diff
@@ -0,0 +1,241 @@
+
+class ArrayDeque[E] private (private var inner: ArrayList[E])
+    extends AbstractCollection[E]
+    with Deque[E]
+    with Cloneable
+    with Serializable {
+  self =>
+
+  private var status = 0
+
+  def this() =
+    this(new ArrayList[E](16))
+
+  def this(initialCapacity: Int) = {
+    // This is the JVM behavior for negative initialCapacity.
+    this(new ArrayList[E](Math.max(0, initialCapacity)))
+  }
+
+  def this(c: Collection[_ <: E]) = {
+    this(c.size)
+    addAll(c)
+  }
+
+  override def add(e: E): Boolean = {
+    offerLast(e)
+    true
+  }
+
+  def addFirst(e: E): Unit =
+    offerFirst(e)
+
+  def addLast(e: E): Unit =
+    offerLast(e)
+
+  // shallow-copy
+  override def clone(): ArrayDeque[E] =
+    new ArrayDeque[E](inner.clone.asInstanceOf[ArrayList[E]])
+
+  def offerFirst(e: E): Boolean = {
+    if (e == null) {
+      throw new NullPointerException()
+    } else {
+      inner.add(0, e)
+      status += 1
+      true
+    }
+  }
+
+  def offerLast(e: E): Boolean = {
+    if (e == null) {
+      throw new NullPointerException()
+    } else {
+      inner.add(e)
+      status += 1
+      true
+    }
+  }
+
+  def removeFirst(): E = {
+    if (inner.isEmpty)
+      throw new NoSuchElementException()
+    else
+      pollFirst()
+  }
+
+  def removeLast(): E = {
+    if (inner.isEmpty)
+      throw new NoSuchElementException()
+    else
+      pollLast()
+  }
+
+  def pollFirst(): E = {
+    if (inner.isEmpty) null.asInstanceOf[E]
+    else {
+      val res = inner.remove(0)
+      status += 1
+      res
+    }
+  }
+
+  def pollLast(): E = {
+    if (inner.isEmpty) null.asInstanceOf[E]
+    else inner.remove(inner.size - 1)
+  }
+
+  def getFirst(): E = {
+    if (inner.isEmpty)
+      throw new NoSuchElementException()
+    else
+      peekFirst()
+  }
+
+  def getLast(): E = {
+    if (inner.isEmpty)
+      throw new NoSuchElementException()
+    else
+      peekLast()
+  }
+
+  def peekFirst(): E = {
+    if (inner.isEmpty) null.asInstanceOf[E]
+    else inner.get(0)
+  }
+
+  def peekLast(): E = {
+    if (inner.isEmpty) null.asInstanceOf[E]
+    else inner.get(inner.size - 1)
+  }
+
+  def removeFirstOccurrence(o: Any): Boolean = {
+    val index = inner.indexOf(o)
+    if (index >= 0) {
+      inner.remove(index)
+      status += 1
+      true
+    } else
+      false
+  }
+
+  def removeLastOccurrence(o: Any): Boolean = {
+    val index = inner.lastIndexOf(o)
+    if (index >= 0) {
+      inner.remove(index)
+      status += 1
+      true
+    } else
+      false
+  }
+
+  def offer(e: E): Boolean = offerLast(e)
+
+  def remove(): E = removeFirst()
+
+  def poll(): E = pollFirst()
+
+  def element(): E = getFirst()
+
+  def peek(): E = peekFirst()
+
+  def push(e: E): Unit = addFirst(e)
+
+  def pop(): E = removeFirst()
+
+  def size(): Int = inner.size
+
+  private def failFastIterator(startIndex: Int, nex: (Int) => Int) = {
+    new Iterator[E] {
+      private def checkStatus() =
```

Comment:
A method body spanning several lines should be enclosed in `{}`.

Reply:
Done, artifact of original code.